ansible_collections/arista/avd/plugins/module_utils/utils.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+class AristaAvdError(Exception):
+    def __init__(self, message="An Error has occured in an arista.avd plugin"):
+        self.message = message
+        super().__init__(self.message)
+
+
+class AristaAvdMissingVariableError(AristaAvdError):
+    pass
```

Comment:
Aren't we throwing any error?

Reply:
This is wrapping AristaAvdError, and caught in the calling function of the action plugin, where it raises another exception. As far as I know, this is how you wrap more generic exceptions.